homework/81984/src/Tokenizer.cpp

```diff
@@ -0,0 +1,272 @@
+#include "Tokenizer.hpp"
+
+namespace Rb {
+  #define KEYWORD_REGEX               Regex("^[a-z\\?]+")
+  #define IDENTIFIER_REGEX            Regex("^([a-zA-Z_]+)([a-zA-Z0-9_\\?]*)")
+  #define FUNCTION_IDENTIFIER_REGEX   Regex("^([a-zA-Z_]+)([a-zA-Z0-9_\\?]*)\\(")
+  #define SYMBOL_REGEX                Regex("^:(([a-zA-Z_]+)([a-zA-Z0-9_]*)|'.*'|\".*\")")
+  #define SINGLE_QUOTED_STRING_REGEX  Regex("^'.*'")
+  #define DOUBLE_QUOTED_STRING_REGEX  Regex("^\".*\"")
+  #define FUNCTION_IDENTIFIER_REGEX   Regex("^([a-zA-Z_]+)([a-zA-Z0-9_\\?]*)\\(")
+  #define NUMBER_REGEX                Regex("^[+-]?([0-9]*[.])?[0-9]+")
+  #define TRIPLE_DOT_REGEX            Regex("^\\.\\.\\.")
+  #define DOUBLE_DOT_REGEX            Regex("^\\.\\.")
+  #define TRIPLE_EQUAL_REGEX          Regex("^\\=\\=\\=")
+  #define DOUBLE_EQUAL_REGEX          Regex("^\\=\\=")
+  #define GREATER_THAN_OR_EQAUL_REGEX Regex("^>\\=")
+  #define LESS_THAN_OR_EQAUL_REGEX    Regex("^<\\=")
+  #define NO_REGEX_MATCH              String("")
```

Comment:
Can be members of the `Tokenizer` to avoid creating DFA/NFA from the regex on every match